### ISSUE TITLE
UseEnumSwitch - detection edge case (false positive)

### ISSUE THREAD

**Issue by imoverclocked** [google/error-prone#2521]
```java

interface Foo {
  void foo();
}

enum Bar implements Foo {
  Bar1, Bar2, Bar3;

  void foo() { /* do something */ }
}

enum Baz implements Foo {
  Baz1, Baz2, Baz3;

  void foo() { /* do something else */ }
}
```

Both enums implement an interface but code like this can't be turned into an enum switch because they are two different enums:

```java
final Foo quux = getAFoo();
if (quux == Bar.Bar1) { /* some code */ }
else if(quux == Baz.Baz1) { /* some code */ }
else if(quux == Baz.Baz2) { /* some code */ }
```

**Comment by sumitbhagwani** [comment]
You explained it correctly. Since the comparisons are made across different enums, it doesn't make sense for a suggestion for enum switch?

**Comment by imoverclocked** [comment]
Yes, currently it suggests a fix when it shouldn’t.
